File: TPC v2.0 Online Round/python/Minimum Table Size.py
import math
n,p = map(int, input().split())

a = list(map(int, input().split()))
cost = 0
temp  = 0
for i in a:
    if math.ceil(math.log(i,p)) == 0:
        temp += p ** math.ceil(math.log(i,p))+1
    else:
        temp += p ** math.ceil(math.log(i,p))
print(temp)

File: TPC v2.0 Online Round/c++/Minimum Table Size.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
	int n,p;
	cin >> n>>p;
	long long ans=0;
	for(int i=0;i<n;i++){
		int x;
		cin >> x;
		double val = ceil(log10(x) / log10(p));
		if(val == 0){
			ans += pow(p, val+1);
		}else{
			ans += pow(p, val);
		}
	}
	cout << ans;	
}

void usingBinarySearch(){
	const long long oo = 1e6;
	set <long long> powers;
	
	int n,p;
	cin >> n >> p;
	for (long long val= p;p<oo;val*=p){
		powers.insert(val);
	}
	long long cost = 0;
	for(int i=1; i<=n;i++){
		int x;
		cin >> x;
		auto it = upper_bound(powers.begin(), powers.end(), x-1);
		
		cost += *it;
	}
	cout << cost ;
}
